ocrd/ocrd/lib.bash

```diff
@@ -137,6 +140,8 @@ ocrd__parse_argv () {
             -w|--working-dir) ocrd__argv[working_dir]=$(realpath "$2") ; shift ;;
             -m|--mets) ocrd__argv[mets_file]=$(realpath "$2") ; shift ;;
             --overwrite) ocrd__argv[overwrite]=true ;;
+            --profile) ocrd__argv[profile]=true ;;
```

Comment:
I don't think we should elevate this to spec. But having it in core anyway, it's only natural to at least pass the information on in the arg parser for bashlib. Of course there is not so much more we can do from bashlib (except perhaps `set -x`, or even `exec 3>${ocrd__argv[profile_file]}; export BASH_XTRACEFD=3`; or maybe some timestamping in the `DEBUG` trap), because the implementation may not use much of bash and only delegate to other programs. The flag would still be useful then, for example by activating some native profiling mechanism in that program.

Reply:
`exec 3>${ocrd__argv[profile_file]}; export BASH_XTRACEFD=3`

That's an amazing snippet I will save for later :)

> I don't think we should elevate this to spec. 

I agree, so for consistency's sake, shall we change the mechanism in the python implementation to use environment variables (`OCRD_PROFILE`, `OCRD_PROFILE_FILE`) and drop it from the `--help` output?

Reply:
I get that, but consistency with the current state of affairs is also the target of this PR, so I thought we could handle `--profile` vs. `OCRD_PROFILE` here before we start to adding new flags for caching, processing server etc.

Reply:
What do you mean `also`?

This PR brings `--profile` to bashlib, nothing more.

Revisit when there is a general mechanism is all I'm saying

Reply:
> What do you mean `also`?

"The PR wants to align bashlib and python CLI" is what I meant.

I agree with the `--dump-module-dir` addition and implementation. This should be part of the CLI spec, I just forgot about it

> Revisit when there is a general mechanism is all I'm saying

Since we agree that `--profile*` should not be part of the CLI spec, we could get rid of those flags now, rather than introduce them in bashlib.

If we don't want to decide now and revisit later, I would not add the flags to the bashlib so that users are aware that `--profile` and `--profile-file` are not actually implemented by bashlib processors raising an error.

But maybe I am overthinking this and we should just merge as-is and revisit later :-D

Reply:
> Since we agree that `--profile*` should not be part of the CLI spec, we could get rid of those flags now, rather than introduce them in bashlib.

Oh, now I got it. But that's a breaking change! And I doubt we will arrive at a definitive conclusion for the general configuration/customization issue quickly. So why not just get this done first?

> If we don't want to decide now and revisit later, I would not add the flags to the bashlib so that users are aware that `--profile` and `--profile-file` are not actually implemented by bashlib processors raising an error.

Ah, I see! Then let's just implement them quick and dirty (see above) _as a first step_.

Also, with a thing like bashlib, you cannot say that they are not implemented: bashlib processors can now start implementing them. (But not without the parser!)

Reply:
So this is still just _tracing_ (but again, the processor code can act on `ocrd__argv[profiling]`  and do something else.

Reply:
Yes, it will also give us hints which parts of the bashlib processors need to be streamlined or parallelised. And we can just ask the user to provide these logs from their environment now.